### ISSUE TITLE
logging: detectKubernetesResource() should set region not zone for location

### ISSUE THREAD

**Issue by tamayika** [googleapis/google-cloud-go#9070]
**Client**

Logging

**Environment**

on GKE

**Go Environment**

N/A

**Code**

N/A

**Expected behavior**

detectKubernetesResource() set region for location

**Actual behavior**

detectKubernetesResource() set zone for location

**Screenshots**

N/A

**Additional context**

`detectKubernetesResource()` currenly set zone(like "us-central1-b") for `location`.
But default stackdriver exporter in GKE uses region(like "us-central1") for `location` and GCP UI generates Cloud Logging URL which uses region for `location` at Pod logs page.
So, when developers want to switch from stackdriver exporter to direct exporter by this library, location value is not compatible between changes and default detection is not useful.

**Comment by minherz** [comment]
@tamayika can you please clarify what do you call "default stackdriver exporter" and to share an example of the resource metadata from the logs that are ingested by the default stackdriver exporter?

thank you

**Comment by tamayika** [comment]
This is the example of metadata which default stackdriver exporter uploads.(i.e. stdout or stderr logs metadata)

```yaml
resource:
  labels:
    cluster_name: "my-cluster"
    container_name: "my-container"
    location: "us-central1"
    namespace_name: "my-namespace"
    pod_name: "my-pod"
    project_id: "my-project"
  type: "k8s_container"
```

This is the example of metadata which current logging library uploads.
```yaml
resource:
  labels:
    cluster_name: "my-cluster"
    container_name: "my-container"
    location: "us-central1-b"
    namespace_name: "my-namespace"
    pod_name: "my-pod"
    project_id: "my-project"
  type: "k8s_container"
```

So, only `location` is different.

**Comment by gkevinzheng** [comment]
@tamayika according to documentation for the resource metadata for `k8s_container` types (https://cloud.google.com/monitoring/api/resources#tag_k8s_container), the value for `location` can be either a region or a zone. While I agree with you that it is unclear/ambiguous what value should go into `location`, the default stackdriver exporter is a legacy item that we will not support with an integration such as this one that will break the existing experience for other customers as well.

**Comment by tamayika** [comment]
Thanks for consideration.
I will propose to change GKE logs URL at GCP Support Page.

**Comment by arekkusu** [comment]
@gkevinzheng The doc you mentioned regarding the location label (for `k8s_container`) says:

>  The physical location of the cluster that contains the container.

To be this means that location should be set to a zone in the case of a GKE zonal cluster and to the region in the case of a GKE regional cluster. 

It seems this is the exact behavior that happens when going through the default GKE logging pipeline. I'm not sure what 
 "stackdriver exporter" / legacy is meant to refer to exactly.

I think the reported issue here is that the logging library doesn't set "location" to the Cluster zone or region but the zone where the node exists.

**Comment by gkevinzheng** [comment]
@arekkusu Ahh, this makes sense. I'll add logic to the library for distinguishing those two use-cases then.

**Comment by minherz** [comment]
@tamayika thank you for [providing](https://github.com/googleapis/google-cloud-go/issues/9070#issuecomment-1839889969) data. I am right to assume that by "default stackdriver exporter" your reference the implicit logging agent of GKE?

**Comment by minherz** [comment]
According to the current implementation:
<https://github.com/googleapis/google-cloud-go/blob/286d7f33e4cbeb5dc35ef365d3838db5524b6ad1/logging/resource.go#L199-L205>

the "location" attribute value is correctly retrieved from the metadata server using the `http://metadata.google.internal/computeMetadata/v1/instance/attributes/cluster-location` URL.

@tamayika can you please confirm that you use the latest version of the library?

**Comment by tamayika** [comment]
@minherz 

> the "location" attribute value is correctly retrieved from the metadata server using the http://metadata.google.internal/computeMetadata/v1/instance/attributes/cluster-location URL.

Its fix is in [v1.9.0](https://github.com/googleapis/google-cloud-go/releases/tag/logging%2Fv1.9.0) which is released several hours ago.
I tried v1.9.0 and confirmed it worked as I expected.
